base_layer/core/src/base_node/rpc/sync_utxos_by_block_task.rs

```diff
@@ -158,16 +158,17 @@ where B: BlockchainBackend + 'static
                 current_header.height,
                 current_header_hash.to_hex(),
             );
-
-            let utxo_block_response = SyncUtxosByBlockResponse {
-                outputs: utxos,
-                height: current_header.height,
-                header_hash: current_header_hash.to_vec(),
-                mined_timestamp: current_header.timestamp.as_u64(),
-            };
-            // Ensure task stops if the peer prematurely stops their RPC session
-            if tx.send(Ok(utxo_block_response)).await.is_err() {
-                break;
+            for utxo_chunk in utxos.chunks(2000) {
```

Comment:
We can safely increase this number to 4000 as 5526 UTXOs in the faucet takes up approximately 4.5924 MiB. Thus (4000 / 5526) x 4.5924 = 3.3242 MiB.